ports/psoc6/boards/CY8CPROTO-062-4343W/makefile_mtb.mk

```diff
@@ -45,59 +46,51 @@ mpy_mtb_init:
+#	$(info MPY_MTB_LIBRARIES : $(MPY_MTB_LIBRARIES))
+	$(eval LIBS += $(MPY_MTB_LIBRARIES) $(MPY_MTB_BOARD_BUILD_OUTPUT_DIR)/$(MPY_MTB_LIB_NAME))
+#	$(info LIBS : $(LIBS))
 	$(eval CFLAGS += -mcpu=cortex-m4 --specs=nano.specs -Og -mfloat-abi=softfp -mfpu=fpv4-sp-d16 -mthumb -ffunction-sections -fdata-sections -ffat-lto-objects -g -Wall -pipe -DCOMPONENT_4343W -DCOMPONENT_APP_CY8CPROTO_062_4343W -DCOMPONENT_CAT1 -DCOMPONENT_CAT1A -DCOMPONENT_CM0P_SLEEP -DCOMPONENT_CM4 -DCOMPONENT_CM4_0 -DCOMPONENT_Debug -DCOMPONENT_GCC_ARM -DCOMPONENT_HCI_UART -DCOMPONENT_MURATA_1DX -DCOMPONENT_MW_CAT1CM0P -DCOMPONENT_MW_CMSIS -DCOMPONENT_MW_CORE_LIB -DCOMPONENT_MW_CORE_MAKE -DCOMPONENT_MW_MTB_HAL_CAT1 -DCOMPONENT_MW_MTB_PDL_CAT1 -DCOMPONENT_MW_RECIPE_MAKE_CAT1A -DCOMPONENT_MW_RETARGET_IO -DCOMPONENT_PSOC6_02 -DCOMPONENT_SOFTFP -DCOMPONENT_WIFI_INTERFACE_SDIO -DCORE_NAME_CM4_0=1 -DCY8C624ABZI_S2D44 -DCY_APPNAME_mtb_example_hal_hello_world -DCY_SUPPORTS_DEVICE_VALIDATION -DCY_TARGET_BOARD=APP_CY8CPROTO_062_4343W -DCY_USING_HAL -DDEBUG -DTARGET_APP_CY8CPROTO_062_4343W)
-	$(info )
+#	$(info CFLAGS : $(CFLAGS))
 	$(eval CXXFLAGS += $(CFLAGS))
-	$(info )
-	$(eval LDFLAGS += -mcpu=cortex-m4 --specs=nano.specs -mfloat-abi=softfp -mfpu=fpv4-sp-d16 -mthumb -ffunction-sections -fdata-sections -ffat-lto-objects -g -Wall -pipe -Wl,--gc-sections -T$(MPY_DIR_OF_MTB_ADAPTER_MAKEFILE)/bsps/TARGET_APP_CY8CPROTO-062-4343W/COMPONENT_CM4/TOOLCHAIN_GCC_ARM/linker.ld -Wl,-Map,$(BUILD)/firmware.map $(MPY_MTB_OBJECTS) -Wl,--start-group -Wl,--end-group -Wl,--print-memory-usage)
-	$(info )
-	$(info LD : $(LD))
-	$(info )
+#	$(info CXXFLAGS : $(CXXFLAGS))
+	$(eval LDFLAGS += -mcpu=cortex-m4 --specs=nano.specs -mfloat-abi=softfp -mfpu=fpv4-sp-d16 -mthumb -ffunction-sections -fdata-sections -ffat-lto-objects -g -Wall -pipe -Wl,--gc-sections -T$(MPY_DIR_OF_MTB_ADAPTER_MAKEFILE)/bsps/TARGET_APP_CY8CPROTO-062-4343W/COMPONENT_CM4/TOOLCHAIN_GCC_ARM/linker.ld -Wl,-Map,$(BUILD)/firmware.map -Wl,--start-group -Wl,--end-group -Wl,--print-memory-usage)
+#	$(info LDFLAGS : $(LDFLAGS))
 	$(eval QSTR_GEN_CFLAGS += $(INC))
 	touch mpy_define_mtb_vars
 
 
-mpy_program: $(MPY_MAIN_BUILD_DIR)/firmware.elf
+# the following can be used without usbip
+mpy_program_win: $(MPY_MAIN_BUILD_DIR)/firmware.hex
 	$(info )
-	$(info Replacing firmware.hex MTB build folder)
-	$(Q) cp $(MPY_MAIN_BUILD_DIR)/firmware.elf $(MPY_MTB_BOARD_BUILD_OUTPUT_DIR)/$(MPY_MTB_APPNAME).elf
-	$(info Programming using MTB programmer)
-	$(Q) cd $(BOARD_DIR); $(MAKE) program
+	$(info Programming using openocd ...)
+	$(OPENOCD_WIN_HOME)/bin/openocd.exe -s $(OPENOCD_WIN_HOME)/scripts -s bsps/TARGET_APP_CY8CPROTO-062-4343W/config/GeneratedSource -c "source [find interface/kitprog3.cfg]; ; source [find target/psoc6_2m.cfg]; psoc6 allow_efuse_program off; psoc6 sflash_restrictions 1; program $(MPY_DIR_OF_MAIN_MAKEFILE)/build/firmware.hex verify reset exit;"
+	$(info Programming done.)
```

Comment:
What is the difference between the 2 recipes but the paths? and the ".exe" extension? Can they be combined in one? en extension can be a variable empty for unix? 

In general, given that no real support is there for windows for the complete workflow... I am not sure if the benefits of flashing from windows are worth making the makefile more complex.

Reply:
I think combining the 2 targets would make it more complex. 
This is just to be able to use Windows instead of WSL flashing in case WSL hangs. And that happens from time to time. ANd always when you do not need it.

Reply:
Yes, okay. but it is very specific to our case... I would expect any contributor to have access to a linux machine without that many obstacles.

Reply:
Agree with both. Keeping them separate for now is justified. We may not need the Windows target later.

Reply:
I think I did not mention that before : Having moved from an application (APPNAME in Makefile) to a library (LIBNAME), MTB disables programming as there is no main routine.  The information, ie command line, to flash should also be provided by MTB just like all the other options (CFLAGS, LDFLAGS, ...). In order not to clutter up the main part of the makefile_mtb.mk we could just put the windows stuff and respective variable settings into a separate file that is included. What do you think ?

Reply:
what does it mean "MTB disables programming". You mean the " make program" target is not available in the modus toolbox? 
But, yes, maybe all these windows special targets could be in a separate file. It would be nice not to bring a lot of "special cases" into the mpy ecosystem. I think no port support development from windows natively.

Reply:
right, make program is disabled when building a library. From the MTB point of view that makes sense.

Reply:
We could have our "flashing etc" targets in  a separate makefile and include it. Then we are flexible with both windows and linux options.